configuration/mongo-configuration.ts

```diff
@@ -0,0 +1,124 @@
+import {MongoClient, Db, Collection} from "mongodb";
+import {IConfiguration} from "./iconfiguration";
+
+export class MongoConfiguration implements IConfiguration {
+    /**
+     * Number of seconds to wait between each attempt at connecting
+     * to the DB.
+     */
+    private static timeoutSeconds: number = 8;
+    private mongoConfig: { [key: string]: any } = {};
+    private requiredKeys: string[];
```

Comment:
Minimize the amount of class variables: things like requiredKeys are only needed during initialization, there's no need to hold on to this data for the entirety of the application.